docs/security/index.md

```diff
@@ -0,0 +1,453 @@
+existing deployments to continue to function as before as applications are migrated onto the new Security APIs.
+
+See the `security-rdf-plugin-abac` and `security-plugin-rdf-abac-registration` module for the logic and registration
+modules per [Separating Logic and Registration](#separating-logic-and-registration).
+
+## Dependencies
+
+For application developers they **MUST** only depend on the `security-core` module as a `compile` scoped dependency i.e.
+
+```xml
+<dependency>
+  <groupId>io.telicent.smart-caches</groupId>
+  <artifactId>security-core</artifactId>
+  <version>X.Y.Z</version>
+</dependency>
+```
+
+Where `X.Y.Z` is the latest [release](../../README.md#depending-on-these-libraries) of these libraries.
+
+### Test Dependencies
+
+For `test` scope where an application *MAY* require a concrete implementation of a `SecurityPlugin` in order to write
+security related tests then you **SHOULD** depend on the [Default Plugin](#default-plugin) as a test dependency i.e.
+
+```xml
+<dependency>
+  <groupId>io.telicent.smart-caches</groupId>
+  <artifactId>security-plugin-rdf-abac-registration</artifactId>
+  <version>X.Y.Z</version>
+  <scope>test</scope>
+</dependency>
+```
+
+This ensures that any security related tests are verified with our default plugin.  For test purposes application
+developers *MAY* also wish to provide mechanisms that allow them to supply a `SecurityPlugin` implementation for testing
+without going through [`SecurityPluginLoader.load()`](#plugin-loading).  This allows them to provide mock, or other test
+implementations, of the API to provoke specific error handling conditions/scenarios that are necessary to achieve
+appropriate test coverage.
+
+## Implementation Concerns
+
+### Separating Logic and Registration
+
+As described [earlier](#plugin-loading) the system expects `1`, and only `1`, plugin to be registered.  At the same time
+it is also designed to allow for multiple label and entitlement schemas to potentially co-exist in a single plugin.
+
+This means that generally implementations should be done as two modules:
+
+- A Logic module that provides the actual interface implementations that implement a particular security model/policy
+  framework within the API.
+- A Registration module that depends on the logic module and provides the appropriate `META-INF/services` file to
+  register the plugin.
+
+By implementing things in this way existing implementations of different label schemas can be combined by creating a new
+plugin that depends on the logic modules of one/more existing plugin implementations, with its implementations wrapping
+those existing implementations.  This can then be registered as a combined plugin via its own registration module.
+
+### Caching for Performance
+
+In real deployments the actual pool of distinct labels is often relatively small.  Therefore implementations can
```

Comment:
Is this true? We don't have any fully operational customer deployments yet. The features ODY are putting forward imply this won't necessarily be the case.

Reply:
I think it's only really true for our internal development clusters, for actual customer deployments there seems to be far larger variation in labelling so need to rephrase this section to make it clearer

Reply:
Section has been reworded accordingly